src/representation/dmp/src/QuaternionDmpModel.h

```diff
@@ -16,7 +16,7 @@ class QuaternionDmpModel : public dmp_cpp::DMPModel{
   QuaternionDmpModel(const DMPModel& other) : DMPModel(other)
   {  }
 
-  virtual bool is_valid() const
+  bool is_valid() const
```

Comment:
This does not work because QuaternionDmpModel is a subclass of DMPModel and `ìs_valid` is defined in DMPModel. Can you copy the content of DMPModel to QuaternionDmpModel and remove the inheritance?

Reply:
What do you mean with "does not work". Does it compile?

Reply:
Yes, but it does not work as intended, simple example:

```c++
#include <iostream>

struct A
{
  void fun() { std::cout << "A" << std::endl; }
};
struct B : public A
{
  void fun() { std::cout << "B" << std::endl; }
};


int main()
{
  A* b = new B();
  b->fun();
  delete b;
}
```
Output:
```
A
```

(Could also result in a memory leak because the destructor is not virtual.)

Reply:
True, but copying all the content is a bit ugly ain' t it? 

The reference to the base class' is_valid() method is used nowhere, where actually the derived class' is_valid() should be called. To avoid confusion, one could rename the is_valid() from QuarternionDMPModel. What do you think?

Reply:
If you don't like to copy everything from the base class, can you please document explicitely that `is_valid` is not virtual? In the base class and in the derived class?